pkg/terraform/terraform.go

```diff
@@ -0,0 +1,190 @@
+	Plan(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	Apply(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	Show(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error)
+	GetEntrypointDirectories(rootDir string) ([]string, error)
+}
+
+// TerraformClient implements the Terraform interface.
+type TerraformClient struct{}
+
+// NewTerraformClient creates a new Terraform client.
+func NewTerraformClient() *TerraformClient {
+	return &TerraformClient{}
+}
+
+// Init runs the terraform init command.
+func (t *TerraformClient) Init(ctx context.Context, workingDir string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"init"}
+	childArgs = append(childArgs, args...)
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// Validate runs the terraform validate command.
+func (t *TerraformClient) Validate(ctx context.Context, workingDir string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"validate"}
+	childArgs = append(childArgs, args...)
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// Show runs the Terraform show command.
+func (t *TerraformClient) Show(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"show"}
+	childArgs = append(childArgs, args...)
+
+	if file != "" {
+		childArgs = append(childArgs, file)
+	}
+
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// Plan runs the Terraform plan command.
+func (t *TerraformClient) Plan(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"plan"}
+	childArgs = append(childArgs, args...)
+
+	if file != "" {
+		childArgs = append(childArgs, fmt.Sprintf("-out=%s", file))
+	}
+
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// Apply runs the Terraform apply command.
+func (t *TerraformClient) Apply(ctx context.Context, workingDir, file string, args ...string) ([]byte, []byte, int, error) {
+	childArgs := []string{"apply"}
+	childArgs = append(childArgs, args...)
+
+	if file != "" {
+		childArgs = append(childArgs, file)
+	}
+
+	return child.Run(ctx, workingDir, "terraform", childArgs) //nolint:wrapcheck
+}
+
+// GetEntrypointDirectories gets all the directories that have Terraform config
+// files containing a backend block to be used as an entrypoint module.
+func (t *TerraformClient) GetEntrypointDirectories(rootDir string) ([]string, error) {
+	matches := make(map[string]struct{})
+	if err := filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to walk directory %s: %w", path, err)
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		matched, err := filepath.Match("*.tf", filepath.Base(path))
```

Comment:
You might like [filepath.Ext](https://pkg.go.dev/path/filepath#Ext) to simplify this a bit